com.vrcfury.vrcfury/SPS/sps_main.cginc

```diff
@@ -58,18 +58,34 @@ void sps_apply_real(inout float3 vertex, inout float3 normal, uint vertexId, ino
 			applyLerp = min(applyLerp, 1-entranceAngleTooSharp);
 		}
 		
+		const float hiltedSphereRadius = 0.3;
 		if (!isRing) {
 			// Uncancel if hilted in a hole
-			const float hiltedSphereRadius = 0.3;
 			const float inSphere = orfDistance > worldLength*hiltedSphereRadius ? 0 : 1;
 			//const float hilted = min(isBehind, inSphere);
 			//shrinkLerp = hilted;
 			const float hilted = inSphere;
 			applyLerp = max(applyLerp, hilted);
 		} else {
-			// Cancel if ring is near or behind base
-			const float isBehind = rootPos.z > 0 ? 0 : 1;
-			applyLerp = min(applyLerp, 1-isBehind);
+			// Align to axis if hilted in a ring
+			// Prevent orifice from getting too close to plug as that can introduce excessive roll
+			const float minDistance = worldLength * 0.2;
+			const float lerpDistance = minDistance;
+			
+			// Consider plane aligned to orifice
+			const float orfPerpDistance = dot(sps_normalize(-frontNormal), rootPos);
```

Comment:
Why is this beneficial vs just using orfDistance directly?

Reply:
This was done for hypothetical hilted/uni-directional rings.

If we use orfDistance directly, we could end-up lerping out the min distance logic if the socket goes behind the plug, causing the plug to suddenly pull back.

Reply:
Including an sketch of the two
![Drawing1 p1_cr](https://github.com/VRCFury/VRCFury/assets/4604733/e042bec3-599c-4f85-a555-7fc90418702f)

Where the dashed line shows the path of the plug as it moves pass/though the socket
I wanted to avoid the plug starting to lerp towards the original socket as it reaches the far end of the circled area.